otherlibs/systhreads/st_stubs.c

```diff
@@ -408,9 +409,16 @@ static void caml_thread_reinitialize(void)
      effort. */
   if (st_masterlock_init(m) != 0)
     caml_fatal_error("Unix.fork: failed to reinitialize master lock");
-  /* FIXME: The same should be done (or not) for IO mutexes (as in the
-     pre-multicore world). However there they might be locked by
-     someone else, and we are probably in an inconsistent state. */
+
+  /* Reinitialize IO mutexes, in case the fork happened while another thread
+     had locked the channel. If so, we're likely in an inconsistent state,
+     but we may be able to proceed anyway. */
+  caml_plat_mutex_init(&caml_all_opened_channels_mutex);
```

Comment:
`caml_all_opened_channels_mutex` was introduced in OCaml 5 for the case of concurrent access by two domains. For single-domain programs, `caml_all_opened_channels` is protected by the domain lock. Since `fork` is only for single-domain programs, I think `caml_all_opened_channels_mutex` is always in an unlocked state and its data is valid. Thus, maybe there is no need to reinitialise it?

Reply:
I see, so in a single domain program, `caml_all_opened_channels_mutex` can't be locked at the fork because it's never locked during a blocking section. Is that right?

(Deleted this change)

Reply:
Yes, or I would see it as a bug with OCaml 5 (there is no such mutex in OCaml 4).